packages/shared/src/contexts/payment/index.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import type { ReactElement } from 'react';
+import dynamic from 'next/dynamic';
+import { isIOSNative } from '../../lib/func';
+import { PaddleSubProvider } from './Paddle';
+import type { PaymentContextProviderProps } from './context';
+
+const StoreKitSubProvider = dynamic(() =>
+  import('./StoreKit').then((mod) => mod.StoreKitSubProvider),
+);
+
+export const PaymentContextProvider = ({
+  children,
+}: PaymentContextProviderProps): ReactElement => {
+  if (isIOSNative()) {
+    return <StoreKitSubProvider>{children}</StoreKitSubProvider>;
+  }
+
+  return <PaddleSubProvider>{children}</PaddleSubProvider>;
```

Comment:
Shouldn't we load this dynamic then as well in case you iOS we can prevent that massive bundle from loading? (although probably already excluded anyway I guess)

Reply:
All tests failed when I lazy loaded both, plus the bundle size actually increased :lolsob:

Reply:
Oh wtf XD

Reply:
🤷

Reply:
![image](https://github.com/user-attachments/assets/b617232f-dece-44c2-bffe-a1c0e05a59df)